app/support/order_details/cost.rb

```diff
@@ -0,0 +1,20 @@
+class OrderDetails::Cost
+  include ActionView::Helpers::NumberHelper
+
+  def initialize(order_details)
+    @order_details = order_details
+  end
+
+  def final_total
```

Comment:
The one weird thing is, in `ReceiptsController#receipt` it selects order details that are viewable by the acting user. So to get this same behavior in `Order#total_cost` I'll add an optional parameter to optionally filter the order details by an acting user too.

Does that make sense?

```
  def receipt
    @order = Order.find(params[:id])
    raise ActiveRecord::RecordNotFound unless @order.purchased?

    @order_details = @order.order_details.select{|od| od.can_be_viewed_by?(acting_user) }
    raise ActiveRecord::RecordNotFound if @order_details.empty?

    @order_details_cost = OrderDetails::Cost.new(@order_details)

    @accounts = @order_details.collect(&:account).uniq
  end
```